FiftyOne.Pipeline.CloudRequestEngine/FlowElements/CloudAspectEngineBase.cs

```diff
@@ -169,20 +169,33 @@ public override IList<IAspectPropertyMetaData> Properties
                     {
                         if (_aspectProperties == null)
                         {
-                            if (LoadAspectProperties(
-                                RequestEngine.GetInstance()) == false)
-                            {
-                                throw new PipelineException(string.Format(
-                                    CultureInfo.InvariantCulture, 
-                                    Messages.ExceptionFailedToLoadProperties,
-                                    ElementDataKey));
+                            var engineInstance = RequestEngine.GetInstance();
+                            try
+                            {
+                                if (LoadAspectProperties(engineInstance) == false)
+                                {
+                                    throw new PipelineException(string.Format(
+                                        CultureInfo.InvariantCulture,
+                                        Messages.ExceptionFailedToLoadProperties,
+                                        ElementDataKey));
+                                }
+                            }
+                            catch (CloudRequestException ex)
+                            {
+                                throw new PropertiesNotYetLoadedException("Failed to request properties from the cloud.", ex);

```

Comment:
if I understand it correctly - it is not an option to use `PipelineException` with a different `Message` because we don't want to extend `Message`?  or is it because it is impossible to add an inner exception into the `PipelineException`?  just want to understand the motivation for the separate class and so we explain in the comment this motivation.

Reply:
never mind, got the explanation from the above, so please could you add a comment explaining the motivation for using it here, so that we could distinguish this situation from other critical conditions and filter in the catch clause in the calling code in the Pipeline
```
/*
Updated catch clause on the call site to only capture new PropertiesNotYetLoadedException to prevent interference with other critical conditions, e.g. PipelineConfigurationException. Added wrapping logic to CloudAspectEngineBase.Properties.
*/
```